code/__DEFINES/components.dm

```diff
@@ -77,3 +77,17 @@
 // /mob/living/carbon/human signals
 #define COMSIG_HUMAN_MELEE_UNARMED_ATTACK "human_melee_unarmed_attack"			//from mob/living/carbon/human/UnarmedAttack(): (atom/target)
 #define COMSIG_HUMAN_MELEE_UNARMED_ATTACKBY "human_melee_unarmed_attackby"		//from mob/living/carbon/human/UnarmedAttack(): (mob/living/carbon/human/attacker)
+
+//COMPONENT SPECIFIC SIGNALS
+// /datum/component/forensics
+#define COMSIG_FORENSICS_HAS_EVIDENCE "forensics_has_evidence"	//() - returns bitflags as shown below.
+	#define FORENSICS_COMPONENT_HAS_FINGERPRINTS 1
+	#define FORENSICS_COMPONENT_HAS_HIDDENPRINTS 2
+	#define FORENSICS_COMPONENT_HAS_FIBERS 4

```

Comment:
@ninjanomnom in response to your question, I'm going to go with the former. Signals shouldn't be written with the intent of having only one component listen for them. If that's the case, it's better to use the GetComponent family and call procs directly as it's much more efficient. Signals should be for events that several things would feasibly want to listen for. What other than forensics would want to listen for COMSIG_FORENSICS_HAS_EVIDENCE?

Reply:
That there are enough places to want something like this though is the problem. When it's being accessed enough that you want to standardize the access, it starts feeling like it'd be better as a datum.